### PR TITLE
feat: hidden property to collections (#3618)

### LINKED ISSUE #3618: Add an hidden property to collections
**Is your feature request related to a problem? Please describe.**

Anytime you configure a files collection, a corresponding panel is added to the NetlifyCMS view of collections. As a consequence the Collections View of the CMS  quickly become polluted of Collections not meant to be edited (e.g. Read Only) by the Editor and used as fields in other collections.  

**Describe the solution you'd like**

Add a boolean `hidden` property to collections as we already have the same hidden property for fields in such a way that the read-only collections do not pollute the editor view of the collections
 
**Describe alternatives you've considered**

I have no alternative solution at the moment

**Additional context**

When you are migrating a db based CMS to NetlifyCMS, it always happens that you need to make reference to what is known as lookup table in db parlance. Most of the time those lookup tables are readable only by the content editor (e.g. countries, states, cities, etc.). By making files collection (or folder collection as well) hidden-able, we reach a more clean view of collections for the editor.

### PR DESCRIPTION
**Pull request by BA1RY**
Fixes #3618 

**Summary**
Anytime you configure a files collection, a corresponding panel is added to the NetlifyCMS view of collections. As a consequence the Collections View of the CMS quickly become polluted of Collections not meant to be edited (e.g. Read Only) by the Editor and used as fields in other collections.

Therefore, it was decided to add a new boolean property called `hidden` which is `false` by default. When marked `true`, it hides collections which must be read only.

**Test plan**
`yarn test` was run and no errors were detected.
`yarn format` was run to format the code.

Thanks to @magomimmo and @erezrokah for the help.

Non-mandatory cute animal:
puppy

### CODE REVIEW

File: packages/netlify-cms-core/src/components/Collection/Sidebar.js [hunk 1]
```diff
@@ -133,7 +133,12 @@ class Sidebar extends React.Component {
             value={query}
           />
         </SearchContainer>
-        <SidebarNavList>{collections.toList().map(this.renderLink)}</SidebarNavList>
+        <SidebarNavList>
+          {collections
+            .toList()
+            .filter(collection => collection.hidden === false)
```
Comment (erezrokah) [inline packages/netlify-cms-core/src/components/Collection/Sidebar.js:139]: This condition will never be true as. `collection` is an immutablejs instance so you need to access its fields via `collection.get('hidden') !== true`

File: website/content/docs/configuration-options.md [hunk 2]
```diff
@@ -3,13 +3,14 @@ title: Configuration Options
 group: reference
 weight: 23
 ---
+
```
Comment (erezrokah) [inline website/content/docs/configuration-options.md:6]: Can you revert the format changes to this file and only add the `hidden` docs changes?
It makes it very hard to know what had changed.

File: packages/netlify-cms-core/src/components/Collection/Sidebar.js (hunk 1 above)
Comment (BA1RY) [reply packages/netlify-cms-core/src/components/Collection/Sidebar.js:139]: Understood. Will do.

File: website/content/docs/configuration-options.md (hunk 2 above)
Comment (BA1RY) [reply website/content/docs/configuration-options.md:6]: I only added [hidden property in line 200](https://github.com/netlify/netlify-cms/pull/3643/files#diff-1b4d596ba92ef875ca90e980db6f0b98R200) but the prettier formatter changed the format of the document. I'll revert it back.

**Comment by BA1RY** [comment]
@erezrokah I've done the necessary changes and made a new commit for the same.

**Comment by erezrokah** [comment]
@BA1RY thanks again for your work. I made a small change to switch from `hidden` to `hide` to be consistent with other collection options (`create`, `publish`, etc).
Also fixed an issue where if you hide the first collection it would still show it when navigating to the root path and made the docs a bit more clear on the use case.
@magomimmo, @BA1RY  let me know what you think.

**Comment by BA1RY** [comment]
> @BA1RY thanks again for your work. I made a small change to switch from `hidden` to `hide` to be consistent with other collection options (`create`, `publish`, etc).
> Also fixed an issue where if you hide the first collection it would still show it when navigating to the root path and made the docs a bit more clear on the use case.
> @magomimmo, @BA1RY let me know what you think.

Looks good to me. Thanks a lot @erezrokah

**Comment by magomimmo** [comment]
> @BA1RY thanks again for your work. I made a small change to switch from `hidden` to `hide` to be consistent with other collection options (`create`, `publish`, etc).
> Also fixed an issue where if you hide the first collection it would still show it when navigating to the root path and made the docs a bit more clear on the use case.
> @magomimmo, @BA1RY let me know what you think.

I'll check in the afternoon. By reading your comment it sounds good to me too.

**Comment by magomimmo** [comment]
Hi @erezrokah, or I'm doing something wrong or the fix does not work.

Sorry to be pedantic, but I'm a clojurescript/re-frame programmer, not JS/React programmer. I did the following:

* pull the netlify-cms master
* rebuild the library (i.e. `yarn`, `yarn bootstrap` , `yarn build`
* copy `netlify-cms.js` from `packages/netlify-cms/dist/` to `static/admin/js/` 
* change the `src` of the `script` in the `static/admin/index.html` file to the above JS
* run the npx proxy in a terminal
* run `hugo server` from a second terminal

This is the definizion of the files collection with the `hide` option set to `true`:

```yaml
collections:

  # Lookup tables
  - name: 'lookups'
    label: 'Closed Lookup Tables'
    hide: true
    files:
      - label: 'Places'
        name: 'places'
        file: 'data/places.json'
        fields:
          - label: 'Places'
            name: 'places'
            hide: true
            allow_add: false
            widget: 'list'
            fields:
              - label: 'Name'
                name: 'name'
                widget: 'string'
              - label: 'ISTAT ID'
                name: 'id'
                widget: 'string'
```
and this is the the content of `data/places.json` file:

```json
{
  "places": [
    {
      "id": "000002",
      "name": "Bergamo (BG)"
    },
    {
      "id": "000003",
      "name": "Milano (MI)"
    },
    {
      "id": "000004",
      "name": "Lurano (BG)"
    },
    {
      "id": "000005",
      "name": "Roma (RM)"
    },
    {
      "id": "000006",
      "name": "Salerno (SA)"
    },
    {
      "id": "000007",
      "name": "Napoli (NA)"
    },
    {
      "id": "000008",
      "name": "Quarna Sotto (VC)"
    },
    {
      "id": "000009",
      "name": "Verona (VR)"
    }
  ]
}
```

And the new `hide` option it does not seem to work, even if a set it in the `places` file collection as well. Am I worg somewhere?

**Comment by erezrokah** [comment]
The easiest way to test it would probably be:
1. Pull latest master.
2. Run `yarn` and `yarn start`
3. Under `dev-test` directory update the `config.yml` to match the one you're using.
4. Optionally remove all the custom code from `index.html`.

**Comment by magomimmo** [comment]
Thanks @erezrokah. It works! But as said by @hanneskuettner someone with the right URL can still reach the collection.  Is this something we want?

**Comment by erezrokah** [comment]
If someone wants to edit the content they can still fetch the auth token and modify it that way...
We could have more UI based restrictions, but I think we can do it in another PR.

**Comment by magomimmo** [comment]
> We could have more UI based restrictions, but I think we can do it in another PR.

I agree. thanks so much for everything.

**Pull request merged**